src/BilayerNetworks.jl

```diff
@@ -0,0 +1,274 @@
+    elseif isqout(v)
+        return Dict(:label=>"u[$(v-Qoutshift)]'", :style=>"filled", :color=>colors[3])
+    else
+        return Dict(:label=>"$v", :style=>"filled", :color=>colors[4])
+    end
+
+    edge_labeler(e) = if calledge(e)
+        return Dict(:label => edge_labels ? string(e) : "", :color=>"black", :style=>"dashed")
+    elseif effluxedge(e)
+        return Dict(:label => edge_labels ? string(e) : "", :color=>"red")
+    else
+        return Dict(:label => edge_labels ? string(e) : "", :color=>"black")
+    end
+
+    g′ = Catlab.Graphs.Graph()
+    add_parts!(g′, :E, nparts(bn, :Win) + nparts(bn, :Wa) + nparts(bn, :Wn))
+    add_parts!(g′, :V, nparts(bn, :Box) + nparts(bn, :Qin) + nparts(bn, :Qout))
+
+    srcs = vcat(bn[:arg].+Qinshift, bn[:influx], bn[:efflux])
+    tgts = vcat(bn[:call], bn[:infusion].+Qoutshift, bn[:effusion].+Qoutshift)
+
+    g′[:, :src] = srcs
+    g′[:, :tgt] = tgts
+    Catlab.Graphs.PropertyGraph{Any}(g′, node_labeler, edge_labeler;
+                                     prog = prog,
+                                     graph = merge!(Dict(:rankdir => "TB"), graph_attrs),
+                                     node = merge!(Graphics.GraphvizGraphs.default_node_attrs(node_labels), node_attrs),
+                                     edge = merge!(Dict(:arrowsize => "0.5"), edge_attrs),
+                                     )
+end
+
+function GraphvizGraphs.to_graphviz(g::AbstractBilayerNetwork; kwargs...)
+    to_graphviz(propertygraph(g; kwargs...))
+end
+function propertygraph(bn::AbstractLabelledBilayerNetwork;
+                       prog::AbstractString="dot", graph_attrs::AbstractDict=Dict(),
+                       node_attrs::AbstractDict=Dict(), edge_attrs::AbstractDict=Dict(),
+                       node_labels::Bool=true, edge_labels::Bool=false)
+
+    Qinshift = nparts(bn, :Box)
+    Qoutshift = nparts(bn, :Box) + nparts(bn, :Qin)
+
+    colors = ["#E28F41", "#6C9AC3", "#80BE63", "#A8DCD9", "#E0E5CD"]
+
+    # vertex partition helpers V = coproduct(Box, Qin, Qout)
+    isbox(v) = v <=Qinshift
+    isqin(v) = Qinshift < v <= Qoutshift
+    isqout(v) = Qoutshift < v
+
+    # edge partition helpers E = coproduct(Win, W, Wn)
+    calledge(e) = e <= nparts(bn, :Win)
+    influxedge(e) = nparts(bn, :Win) < e <= nparts(bn, :Win) + nparts(bn, :Wa)
+    effluxedge(e) = nparts(bn, :Win) + nparts(bn, :Wa) < e
+
+    node_labeler(v) = if isbox(v)
+        Dict(:label=>"$(bn[v, :parameter])", :style=>"rounded, filled", :width=>"0.75", :height=>"0.5", :fixedsize=>"true", :shape=>"rectangle", :color=>colors[1])
+    elseif isqin(v)
+        return Dict(:label=>"$(bn[v-Qinshift, :variable])", :style=>"filled", :color=>colors[2])
+    elseif isqout(v)
+        return Dict(:label=>"$(bn[v-Qoutshift, :tanvar])'", :style=>"filled", :color=>colors[3])
+    else
+        return Dict(:label=>"$v", :style=>"filled", :color=>colors[4])
```

Comment:
This was a fallback to make sure it didn't fail with a key not found exception

Reply:
Gotcha. This is the last piece that's lacking coverage. Could I replace this with throwing an error?

Reply:
Haha, I guess proper TDD says that if we have error handling logic, then we should test that error handling. I think it is better to test it by making a bilayer with extra vertices rather than to limit the capability of the software. This is in visualization code so a bug that produces extra vertices will manifest as extra vertices with a raw numeric label and color 4. I think that is better than an exception, but could be convinced otherwise.

Reply:
Oh, if we're able to make a bilayer network that triggers this case, that'd be the ideal! I'll take a look at that